include/deal.II/lac/scalapack.templates.h

```diff
@@ -24,6 +24,10 @@
 #include <deal.II/base/mpi.h>
 #include <deal.II/base/mpi.templates.h>
 
+#  ifdef DEAL_II_WITH_HDF5
+#include <hdf5.h>
```

Comment:
actually, you should move the header and those functions below to `scalapack.cc` file. 

`scalapack.templates.h` is reserved for Fortran wrappers. On another hand you don't need those in `scalapack.h` either as it's only the implementation where these functions are used.